### PR TITLE
feat: update implementation guide

### PR DESCRIPTION
**Pull request by paulbalaji**
resolves https://github.com/hyperlane-xyz/issues/issues/950

TODO:
- [x] add cosmos links
- [x] add sealevel links

### CODE REVIEW

File: docs/guides/implementation-guide.mdx [hunk 1]
```diff
@@ -361,22 +408,20 @@ then the message will be kicked to an exponential backoff retry queue. The relay
...
-5. Observe that upon dispatch of a message of the origin chain, the validator observes the message, creates a signature and the relayer appropriately processes your message via the ISM that specifies the validator on the destination chain.
+Once you have implemented an MVP of the relayer, you should create an end-to-end test that:
 
+1. Spins up local origin and destination chains.
+2. Deploys your contracts onto both chains.
+3. Run validators for the origin chain.
+4. Run a relayer between both chains.
+5. Observe that upon dispatch of a message of the origin chain, the validator observes the message, creates a signature and the relayer appropriately processes your message via the ISM that specifies the validator on the destination chain.
 
 See this [end-to-end test on the Rust codebase ](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/utils/run-locally/src/main.rs)for inspiration.
```
Comment (paulbalaji) [inline docs/guides/implementation-guide.mdx:424]: from ticket:
> one definite change in v3 is we added the requirement for e2e test + remote testnet test

not sure what exactly this remote testnet test is, will sync

Comment (paulbalaji) [reply docs/guides/implementation-guide.mdx:424]: update: it means adding some text/instructions around making sure it works on a testnet

File: docs/guides/implementation-guide.mdx [hunk 2]
```diff
@@ -272,25 +346,34 @@ function payForGas(
...
 
 ```solidity
-event GasPayment(
-    bytes32 messageId,
-    uint256 gasAmount,
-    uint256 payment
+event DestinationGasConfigSet(
+    uint32 remoteDomain, // remote domain
+    address gasOracle,   // gas oracle
+    uint96 gasOverhead   // destination gas overhead
```
Comment (yorhodes) [inline docs/guides/implementation-guide.mdx:357]: please leave `GasPayment` here

File: docs/guides/implementation-guide.mdx [hunk 3]
```diff
@@ -272,25 +346,34 @@ function payForGas(
...
 );
 ```
 
+### BenificiarySet
+
+Emitted when the beneficiary is set.
+
+```solidity
+event BeneficiarySet(address beneficiary);
+```
```
Comment (yorhodes) [inline docs/guides/implementation-guide.mdx:367]: nit: omit

File: docs/guides/implementation-guide.mdx [hunk 4]
```diff
@@ -272,25 +346,34 @@ function payForGas(
...
 
 #### Message Indexing
 
-All agents must index messages from the origin mailbox. In the solidity mailbox, we [emit an event for each message](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/solidity/contracts/Mailbox.sol#L125-L129) dispatched. Other chains may have different ways of surfacing this information, but the agent must be able to get message content reliably and with consistent ordering -- see the [message indexer](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/hyperlane-core/src/traits/indexer.rs) trait.
+All agents must index messages from the origin mailbox. In the solidity mailbox, we [emit an event for each message](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/solidity/contracts/Mailbox.sol#L221-L222) dispatched. Other chains may have different ways of surfacing this information, but the agent must be able to get message content reliably and with consistent ordering -- see the [message indexer](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/hyperlane-core/src/traits/indexer.rs) trait.
```
Comment (yorhodes) [inline docs/guides/implementation-guide.mdx:376]: maybe we should permalink these to avoid the maintenance cost?

**Review by yorhodes — APPROVED** [review]

File: docs/guides/implementation-guide.mdx [hunk 5]
```diff
@@ -8,6 +8,17 @@ A Hyperlane implementation for a new chain architecture is comprised of the foll
...
 Before getting started here, it is recommended to review the protocol documentation.
 
+:::info
+
+If you would like to dive deeper, check out some of the Hyperlane implementations that are currently available:
+
+* [Solidity](https://github.com/hyperlane-xyz/hyperlane-monorepo/tree/main/solidity)
+* [Sway](https://github.com/hyperlane-xyz/fuel-contracts)
+* [CosmWasm](https://github.com/hyperlane-xyz/cosmwasm)
+* [Sealevel](https://github.com/hyperlane-xyz/hyperlane-monorepo/tree/main/rust/sealevel)
```
[the remark this answers is not in the document]
Comment (paulbalaji) [reply docs/guides/implementation-guide.mdx:18]: I would if the examples were smaller, but here we link to a full interface or class

**Pull request merged**